src/unix/signal.c

```diff
@@ -63,6 +74,22 @@ static void uv__signal_global_init(void) {
 }
 
 
+static void uv__signal_global_reinit(void) {
+  /* We can only use signal-safe functions here.
+   * That includes read/write and close, fortunately.
+   * We do all of this directly here instead of resetting
+   * uv__signal_global_init_guard because
+   * uv__signal_global_once_init is only called from uv_loop_init
+   * and this needs to function in existing loops.
+   */
+  uv__close(uv__signal_lock_pipefd[0]);
+  uv__signal_lock_pipefd[0] = -1;
+  uv__close(uv__signal_lock_pipefd[1]);
+  uv__signal_lock_pipefd[1] = -1;
+  uv__signal_global_init();
+}
```

Comment:
I want to propose a small enhancement: can you change `uv__signal_lock_pipefd` to:
```c
static int uv__signal_lock_pipefd[2] = { -1, -1 };
```
And assert in that `uv__signal_global_init()` that both elements are -1?  That catches double reinit in case a uv_once_t doesn't stay in 'completed' status after a fork.